docs/src/main/paradox/client-side/http2.md

```diff
@@ -0,0 +1,74 @@
+# Client-Side HTTP/2 (Preview)
+
+@@@ warning
+Client-Side HTTP/2 support in akka-http is currently available as a preview.
+This means it is ready to be evaluated, but the APIs and behavior are likely to change.
+@@@
+
+@@@ note
+It is recommended to first read the @ref[Implications of the streaming nature of Request/Response Entities](../implications-of-streaming-http-entity.md) 
+and @ref[Host-Level Client-Side API](./host-level.md) sections, as they explain the underlying full-stack streaming 
+concepts, tuning the client settings and HTTPS context and how to handle the Request-Response Cycle, which may be 
+unexpected when coming from a background with non-"streaming first" HTTP Clients.
+@@@
+
+## Enable HTTP/2 support
+
+HTTP/2 can then be enabled through configuration:
+
+```
+akka.http.server.preview.enable-http2 = on
+```
+
+## Create the client 
+
+The Akka HTTP client supports HTTP/2 over TLS, and h2c with prior knowledge (plaintext), but not 
+the HTTP Upgrade mechanism.
+
+### HTTP/2 over TLS
+
+To create a client, use the `Http()` fluent API to connect and use the `http2()` creator:
+
+Scala
+:   @@snip[Http2Spec.scala](/docs/src/test/scala/docs/http/scaladsl/Http2Spec.scala) { #http2Client }
+
+Java
+:   @@snip[Http2Test.java](/docs/src/test/java/docs/http/javadsl/Http2Test.java) { #http2Client }
+
+HTTP/2 over TLS needs [Application-Layer Protocol Negotiation (ALPN)](https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation)
+to negotiate whether both client and server support HTTP/2. The JVM provides ALPN support starting from JDK 8u252.
+Make sure you run a JVM version greater than that.
+
+### h2c with prior knowledge
+
+The other option is to connect and start communicating in HTTP/2 immediately. You must know beforehand the target server
+supports HTTP/2 over plaintext. For this reason this approach is known as h2c with
+[Prior Knowledge](https://httpwg.org/specs/rfc7540.html#known-http) of HTTP/2 support.
+
+To create a client, use the `Http()` fluent API to connect and use the `http2WithPriorKnowledge()` creator:
+
+Scala
+:   @@snip[Http2Spec.scala](/docs/src/test/scala/docs/http/scaladsl/Http2Spec.scala) { #http2ClientWithPriorKnowledge }
+
+Java
+:   @@snip[Http2Test.java](/docs/src/test/java/docs/http/javadsl/Http2Test.java) { #http2ClientWithPriorKnowledge }
+
```

Comment:
perhaps add a h2c via HTTP Upgrade section here, explaining again that negotiating between HTTP/1 and HTTP/2 using the `Upgrade` header mechanism isn't supported for Akka HTTP clients?